src/app/features/location/map/map.component.ts

```diff
@@ -0,0 +1,142 @@
+  iconRetinaUrl,
+  iconUrl,
+  shadowUrl,
+  iconSize: [25, 41],
+  iconAnchor: [12, 41],
+  popupAnchor: [1, -34],
+  tooltipAnchor: [16, -28],
+  shadowSize: [41, 41],
+});
+L.Marker.prototype.options.icon = iconDefault;
+
+@Component({
+  selector: "app-map",
+  templateUrl: "./map.component.html",
+  styleUrls: ["./map.component.scss"],
+})
+export class MapComponent<T extends Entity = Entity> implements AfterViewInit {
+  // TODO this should be configurable
+  private readonly start_location: L.LatLngTuple = [52.4790412, 13.4319106];
+
+  @Input() set marked(coordinates: Coordinates | Coordinates[]) {
+    if (!coordinates) {
+      return;
+    }
+    if (Array.isArray(coordinates)) {
+      this.setMultipleMarkers(coordinates);
+    } else {
+      this.setMarker(coordinates);
+    }
+    if (this.map) {
+      this.showMarkersOnMap();
+    }
+  }
+
+  @Input() set entities(entities: { entity: T; property: string }[]) {
+    if (this.markers && this.map) {
+      this.markers.forEach((marker) => marker.removeFrom(this.map));
+    }
+    this.markers = entities
+      .filter(({ entity, property }) => !!entity?.[property])
+      .map(({ entity, property }) => {
+        const marker = L.marker([entity[property].lat, entity[property].lon]);
+        const tooltip = marker.bindTooltip(entity.toString());
+        marker.on("click", () => {
+          this.entityClick.emit(entity);
+          tooltip.openTooltip();
+        });
+        return marker;
+      });
+    if (this.map) {
+      this.showMarkersOnMap();
+    }
+  }
+
+  private map: L.Map;
+  private marker: L.Marker;
+  private markers: L.Marker[];
+  private clickStream = new Subject<L.LatLng>();
+  // TODO filter out double clicks
+  @Output() mapClick: Observable<Coordinates> = this.clickStream.pipe(
+    debounceTime(200),
+    map((c) => ({ lat: c.lat, lon: c.lng }))
+  );
+
+  @Output() entityClick = new EventEmitter<T>();
+
+  ngAfterViewInit() {
+    this.initMap();
+  }
+
+  private initMap() {
+    this.map = L.map("map", {
+      center: this.marker ? this.marker.getLatLng() : this.start_location,
+      zoom: 14,
+    });
+    this.map.addEventListener("click", (res) =>
+      this.clickStream.next(res.latlng)
+    );
+
+    const tiles = L.tileLayer(
+      "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
+      {
+        maxZoom: 18,
+        minZoom: 3,
+        attribution:
+          '&copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>',
+      }
+    );
+    tiles.addTo(this.map);
+
+    if (this.marker || this.markers) {
+      this.showMarkersOnMap();
+    }
+  }
+
+  private setMultipleMarkers(coordinates: Coordinates[]) {
+    if (this.markers && this.map) {
+      this.markers.forEach((marker) => marker.removeFrom(this.map));
+    }
+    this.markers = coordinates
+      .filter((coord) => !!coord)
+      .map((coord) => L.marker([coord.lat, coord.lon]));
+  }
+
+  private setMarker(coordinates: Coordinates) {
```

Comment:
is it important to handle the single marker differently? Or could we just transform that to an array with just one element?

Reply:
the flyTo could be done if markers.length === 1, for example